apps/cookbook/src/app/page/side-nav/side-nav.component.ts

```diff
@@ -1,20 +1,159 @@
-import { Component, OnInit, EventEmitter, Output, Input } from '@angular/core';
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { filter } from 'rxjs/operators';
+
+import { kebabToTitleCase } from '@kirbydesign/designsystem';
+
+import { routes } from '../../showcase/showcase.routes';
+import { navigationItems } from '../header/header.component';
+
+const KEY_DOWN = 'ArrowDown';
+
+interface SideNavLink {
+  path: string;
+  name: string;
+  active: boolean;
+}
 
 @Component({
   selector: 'cookbook-side-nav',
   templateUrl: './side-nav.component.html',
   styleUrls: ['./side-nav.component.scss'],
 })
 export class SideNavComponent implements OnInit {
+  private allShowcaseRoutes: SideNavLink[];
+  filteredShowcaseRoutes: SideNavLink[][];
+  filter: string = '';
```

Comment:
This can be scoped to the enclosing method instead:
```suggestion
```

Reply:
It is bound to the filter input field (to make the clear button work)